### DISCUSSION TITLE
getBoundingClientRect calculate wrong value on studio mode

### DISCUSSION THREAD

**Discussion by mnlbox** [remotion-dev/remotion#3421] [Q&A] [answered]
Hi,

Thanks for this awesome tool. I used getBoundingClientRect function to calculate some object locations but the values are calculating wrong when I try to use studio command. It happened because the viewport and base div is different on Remotion studio but everything is OK when I run the render command.

Is there any helper function to get the location (top, left, right, bottom) of an element correct on both studio and render?

**Comment by JonnyBurger** [comment]
It's a good question!

Let me get back to it tomorrow

**Comment by JonnyBurger** [comment, marked as answer]
I now created some documentation on how you can do it, and an example: https://www.remotion.dev/docs/measuring

**Comment by mnlbox** [comment]
Thanks @JonnyBurger for quick answer.
I tried to get "top" and "left" (not width and height) of element with the same method but it didn't work. I'll check again today and try to create a reproduce sample if it didn't work again.

One more question:
Is there any chance to have this logic somewhere as a helper function in future?
Maybe in `@remotion/layout-utils` package. 🤔
I'm asking because I guess it can be a nice and useful thing to have.

**Reply by JonnyBurger** [reply]
Did you ensure to update to v4.0.103? This is required to not get zeroes.

**Comment by mnlbox** [comment]
@JonnyBurger I still think there is some issue for the calculation, you can see even the ruler value from "studio" is not same as ruler from browser devtools: 🤔 
![image](https://github.com/remotion-dev/remotion/assets/25859075/55c80287-3ee8-4796-857d-b130965de181)

The top right of the composition shows "1250" for x (or top) on studio ruler while it should be "0" or am I missed something for measuring?

**Reply by JonnyBurger** [reply]
It's not meant to be the same - the rulers show the pixels of the video dimensions!

You are rendering a 4K video it appears so the ruler goes from 0-3840 and 0-2160 respectively.

The canvas can be scaled arbitrary with pinch, Ctrl+Scrollwheel or the "%" underneath the canvas, only at 100% the pixels will match up.

**Comment by mnlbox** [comment]
@JonnyBurger  I prepared a sample repo about the ruler and DOM measuring issue that I'm facing now:

https://github.com/mnlbox/remotion-ruler-and-dom-measuring-issue

I guess it's a bug on studio only because everything is working as expected after render the video. I even tried to use the measurer trick that you described in the docs here:
https://www.remotion.dev/docs/measuring

**Comment by mnlbox** [comment]
@JonnyBurger can you please take a look to the above repo to see if it's even an Remotion issue or not?
Maybe I missed something in my code 🤔
I'm a bit confused now because everything is OK in the rendered video.

**Reply by JonnyBurger** [reply]
Sorry for missing this message!

Ok, so:

- Rulers are confirmed broken, they are fixed in 4.0.108
- I found one issue in your code and in our docs, scale is calculated wrong, it is fixed here: https://github.com/remotion-dev/remotion/pull/3472/files
- The usage of `.x` and `.y` is not right - those values are relative to the viewport, not the Remotion Player top left corner. If you want to get the relative coordinate of that, you also need to call getClientBoundingRect() on the most outer container and subtract that.

With these three combined, I hope this resolves all issues!

**Reply by JonnyBurger** [reply]
There is now also https://www.remotion.dev/docs/use-current-scale!
Updated the measuring docs too: https://www.remotion.dev/docs/measuring

**Reply by mnlbox** [reply]
Woooo I like it. This new hook is so neat. Thanks for that. ❤️ 
But I got the error bellow when I change the current frame on the player or when I play the video.
It's working great when I refresh the Remotion Player. Maybe I'm using it wrong 🤔 

<img width="869" alt="image" src="https://github.com/remotion-dev/remotion/assets/25859075/de542c3c-7f79-4c28-8030-3ead9584e21b">

**Reply by JonnyBurger** [reply]
Hi @mnlbox, I could not reproduce this in my testing.
Did you upgrade everything, including `@remotion/cli`?

Can you also try `rm -rf node_modules/.cache` once and tell me if that resolves the issue?

**Reply by mnlbox** [reply]
Thanks @JonnyBurger yeah after removing the `node_modules` and fresh `pnpm install` everything works now.
I also removed pnpm store cache.

Thanks for this awesome tool.
I'm learning new stuff from Remotion every day and I really like the idea behind that.
Thanks for all of your great works.

**Reply by JonnyBurger** [reply]
@mnlbox Super happy to hear that! 🙌🏼
Thanks a lot for helping shape Remotion!
